File: java-junit/src/test/java/com/coffeemachine/Actionwords.java
package com.coffeemachine;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertFalse;
import static junit.framework.Assert.assertTrue;

public class Actionwords {
    public CoffeeMachine sut = new CoffeeMachine();

    public void iStartTheCoffeeMachine() {
        iStartTheCoffeeMachine("en");
    }

    public void iStartTheCoffeeMachine(String lang) {
        sut.start(lang);
    }

    public void iShutdownTheCoffeeMachine() {
        sut.stop();
    }

    public void messageMessageShouldBeDisplayed(String message) {
        assertEquals(sut.message(), message);
    }

    public void coffeeShouldBeServed() {
        assertTrue(sut.coffeeServed);
    }

    public void coffeeShouldNotBeServed() {
        assertFalse(sut.coffeeServed);
    }

    public void iTakeACoffee() {
        sut.takeCoffee();
    }

    public void iEmptyTheCoffeeGrounds() {
        sut.emptyGrounds();
    }

    public void iFillTheBeansTank() {
        sut.fillBeans();
    }

    public void iFillTheWaterTank() {
        sut.fillTank();
    }

    public void iTakeCoffeeNumberCoffees(int coffeeNumber) {
        while ((coffeeNumber > 0)) {
            iTakeACoffee();
            coffeeNumber = coffeeNumber - 1;
        }
    }

    public void theCoffeeMachineIsStarted() {
        iStartTheCoffeeMachine();
    }

    public void fiftyCoffeesHaveBeenTakenWithoutFillingTheTank() {
        iTakeCoffeeNumberCoffees(30);
        iFillTheBeansTank();
        iEmptyTheCoffeeGrounds();
        iTakeCoffeeNumberCoffees(20);
        iFillTheBeansTank();
        iEmptyTheCoffeeGrounds();
    }

    public void thirtyEightCoffeesAreTakenWithoutFillingBeans() {
        iTakeCoffeeNumberCoffees(37);
        iEmptyTheCoffeeGrounds();
        iFillTheWaterTank();
        iTakeACoffee();
    }
}

File: javascript/jasmine/actionwords.js
var Actionwords = {
  coffeeShouldNotBeServed: function () {
    expect(this.sut.get('coffeeServed')).toBeFalsy();
  },

  iFillTheWaterTank: function () {
    this.sut.fillTank();
  },

  iEmptyTheCoffeeGrounds: function () {
    this.sut.emptyGrounds();
  },

  iFillTheBeansTank: function () {
    this.sut.fillBeans();
  },

  iShutdownTheCoffeeMachine: function () {
    this.sut.stop();

  },

  iTakeCoffeeNumberCoffees: function (coffee_number) {
    while ((coffee_number > 0)) {
      this.iTakeACoffee();
      coffee_number = coffee_number - 1;
    }
  },

  iTakeACoffee: function () {
    this.sut.takeCoffee();
  },

  coffeeShouldBeServed: function () {
    expect(this.sut.get('coffeeServed')).toBeTruthy();
  },

  iStartTheCoffeeMachine: function (lang) {
    this.sut.start(lang);
  },

  messageMessageShouldBeDisplayed: function (message) {
    expect(this.sut.get('message')).toEqual(message || "");
  },

  theCoffeeMachineIsStarted: function () {
    this.iStartTheCoffeeMachine();
  },

  fiftyCoffeesHaveBeenTakenWithoutFillingTheTank: function () {
    this.iTakeCoffeeNumberCoffees(30);
    this.iFillTheBeansTank();
    this.iEmptyTheCoffeeGrounds();
    this.iTakeCoffeeNumberCoffees(20);
    this.iFillTheBeansTank();
    this.iEmptyTheCoffeeGrounds();
  },

  thirtyEightCoffeesAreTakenWithoutFillingBeans: function () {
    this.iTakeCoffeeNumberCoffees(37);
    this.iEmptyTheCoffeeGrounds();
    this.iFillTheWaterTank();
    this.iTakeACoffee();
  }
}